src/ui/page_autoscan.c

```diff
@@ -1,84 +1,92 @@
-	lv_obj_set_style_pad_top(label2, 12, 0);
-	lv_label_set_long_mode(label2, LV_LABEL_LONG_WRAP);
-	lv_obj_set_grid_cell(label2, LV_GRID_ALIGN_START, 1, 3,
-						 LV_GRID_ALIGN_START, 4, 2);
-
-	btn_group_set_sel(&btn_group0, g_setting.autoscan.status);
-	btn_group_set_sel(&btn_group1, g_setting.autoscan.source);
-	return page;
+    create_select_item(arr, cont);
+
+    btn_group_t btn_group;
+    create_btn_group_item(&btn_group0, cont, 3, "Auto Scan", "On", "Last", "Off", "", 0);
+    create_btn_group_item2(&btn_group1, cont, 5, "Default", "Last", "HDZero", "Expansion", "AV In", "HDMI In", " ", 1); // 2 rows
+    create_label_item(cont, "< Back", 1, 3, 1);
+
+    lv_obj_t *label2 = lv_label_create(cont);
+    lv_label_set_text(label2, "*if Auto Scan is 'Last', goggles will default to show last tuned channel");
+    lv_obj_set_style_text_font(label2, &lv_font_montserrat_16, 0);
+    lv_obj_set_style_text_align(label2, LV_TEXT_ALIGN_LEFT, 0);
+    lv_obj_set_style_text_color(label2, lv_color_make(255, 255, 255), 0);
+    lv_obj_set_style_pad_top(label2, 12, 0);
+    lv_label_set_long_mode(label2, LV_LABEL_LONG_WRAP);
+    lv_obj_set_grid_cell(label2, LV_GRID_ALIGN_START, 1, 3,
+                         LV_GRID_ALIGN_START, 4, 2);
+
+    btn_group_set_sel(&btn_group0, g_setting.autoscan.status);
+    btn_group_set_sel(&btn_group1, g_setting.autoscan.source);
+    return page;
 }
 
-
-void autoscan_toggle(int sel)
-{
-	int value = 0;
-	
-	if(sel == 0) {
-		btn_group_toggle_sel(&btn_group0);		
-
-		value = btn_group_get_sel(&btn_group0);
-		if(value == 0){
-			ini_puts("autoscan", "status", "scan", SETTING_INI);
-		}else if (value == 1){
-			ini_puts("autoscan", "status", "last", SETTING_INI);
-		}else{
-			ini_puts("autoscan", "status", "menu", SETTING_INI);
-		}
-		g_setting.autoscan.status = value;
-	}
-	else if(sel < 3) {
-		btn_group_toggle_sel(&btn_group1);	
-		value = btn_group_get_sel(&btn_group1);
-		ini_putl("autoscan", "source", value, SETTING_INI);
-		g_setting.autoscan.source = value;
-	}
+static void page_autoscan_on_click(uint8_t key, int sel) {
+    int value = 0;
+
+    if (sel == 0) {
+        btn_group_toggle_sel(&btn_group0);
+
+        value = btn_group_get_sel(&btn_group0);
+        if (value == 0) {
+            ini_puts("autoscan", "status", "scan", SETTING_INI);
+        } else if (value == 1) {
+            ini_puts("autoscan", "status", "last", SETTING_INI);
+        } else {
+            ini_puts("autoscan", "status", "menu", SETTING_INI);
+        }
+        g_setting.autoscan.status = value;
+    } else if (sel < 3) {
+        btn_group_toggle_sel(&btn_group1);
+        value = btn_group_get_sel(&btn_group1);
+        ini_putl("autoscan", "source", value, SETTING_INI);
+        g_setting.autoscan.source = value;
+    }
 }
+
+page_pack_t pp_autoscan = {
+    .p_arr = {
+        .cur = 0,
+        .max = 4,
+    },
+
+    .create = &page_autoscan_create,
```

Comment:
You don't need to pass the address of the function as giving just the name provides the pointer to the function already.
i.e. without the `&`
```
.create = page_autoscan_create,
```